src/IndexRegistryStorage.sol

```diff
@@ -13,17 +13,19 @@ import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
  */
 abstract contract IndexRegistryStorage is Initializable, IIndexRegistry {
 
-    /// @notice The value that indices of deregistered operators are set to
-    uint32 public constant OPERATOR_DEREGISTERED_INDEX = type(uint32).max;
+    /// @notice The value that is returned when an operator does not exist at an index at a certain block
+    bytes32 public constant OPERATOR_DOES_NOT_EXIST_ID = bytes32(0);
 
     /// @notice The RegistryCoordinator contract for this middleware
     IRegistryCoordinator public immutable registryCoordinator;
 
     /// @notice list of all operators ever registered, may include duplicates. used to avoid running an indexer on nodes
     bytes32[] public globalOperatorList;
 
-    /// @notice mapping of operatorId => quorumNumber => index history of that operator
-    mapping(bytes32 => mapping(uint8 => OperatorIndexUpdate[])) internal _operatorIdToIndexHistory;
+    /// @notice mapping of quorumNumber => operator id => current index
+    mapping(uint8 => mapping(bytes32 => uint32)) public operatorIdToIndex;
+    /// @notice mapping of quorumNumber => index => operator id history for that index
+    mapping(uint8 => mapping(uint32 => OperatorUpdate[])) internal _indexToOperatorIdHistory;
```

Comment:
oooohhhh OK nice, this is the crux of the change, isn't it? structuring the storage different so you can look things up from the index.
but it means not being able to get operator IDs' historic indices easily -- I take it that's OK? Is the main existing use for these type of lookups the swap operations (where only current index matters)?

Reply:
the idea is that we only need that for proofs of custody, and we'll have more sophisticated indexing support by then

Reply:
> the idea is that we only need that for proofs of custody, and we'll have more sophisticated indexing support by then

ah OK but we can still use off-chain indexing and prove inputs are correct for that. Seems fine then.